src/Sulu/Component/Content/Types/Number.php

```
@@ -57,4 +65,68 @@ public function read(NodeInterface $node, PropertyInterface $property, $webspace
 
         return $value;
     }
+
+    public function mapPropertyMetadata(ContentPropertyMetadata $propertyMetadata): PropertyMetadata
+    {
+        /** @var string $propertyName */
+        $propertyName = $propertyMetadata->getName();
+        $mandatory = $propertyMetadata->isRequired();
+
+        $min = $this->getFloatParam($propertyMetadata, 'min');
+        $max = $this->getFloatParam($propertyMetadata, 'max');
+        $multipleOf = $this->getFloatParam($propertyMetadata, 'multiple_of');
```

I would not introduce a new parameter here I would reuse the `step` parameter. I would not support to have a different `step` and a different `multiple_of` parameter in the number type.

---

I think it's a valid example to have `step=1` and `multiple_of=0.01` to allow numbers with two decimals (prices e.g.) but the number field should still use `1` as step

---

I know what you mean and wanted to support with it but I personally would not do it. It can be misconfigured step lower then multiple_of, is step fallbacking or is multple_of of fallbacking, will multiple_of overwrite step when lower, many questions I think is added for a usecase which is mostly not used. And if there is just one value its just clear.
But if the others want also to add it I'm fine with it. /cc @sulu/core-team

---

I would at current state go with `step` and wait for feedback. If this is then requested we still could add `multiple_of` but at current state I would just use the value from `step`.

---

We could just throw an exception if `step` is not a multiple of `multiple_of` - then there's no fallback "magic" ... But `step` shouldn't be used for validation IMO, because if you specify `step` on a html input element, you can still type in numbers between steps, there won't be an error. And in the docs we could clearly specify that `step` is just used for the ui while `multiple_of` is used for validation

---

I'm aware of how html is handling it but as said I would at current state not add an additional parameter for it.

---

dont have a strong opinion here - both options are okay for me

---

/cc @chirimoya

---

@luca-rath I did discuss it with @chirimoya and we keep step and multiple_of. So the PR is ready. Can you check the PHP Lint failings then we could also merge this.